girderformindlogger/models/user.py

```diff
@@ -698,3 +698,17 @@ def _deleteProtoUser(self, doc):
         existing = ProtoUser().findOne(q)
         if existing is not None:
             ProtoUser().remove(existing)
+
+    def setUserName(self, user, userName, save=True):
+        """
+        Change a user's username
+        
+        :param user: The user whose username to change.
+        :param userName: the new userName to be stored
+        """
+
+        if len(userName) > 0:
+            user['login'] = userName
+        else:
+            raise Exception('username can\'t be empty')
+        self.save(user)
```

Comment:
This validation step https://github.com/ChildMindInstitute/mindlogger-app-backend/blob/ac4530037ec61f3dc08c626f6bff29213fce7d6e/girderformindlogger/models/user.py#L72-L78 will cause failures since `firstName` and `lastName` are no longer required fields:

![validation failure for user with no `firstName`](https://user-images.githubusercontent.com/5974438/74449847-2ba57980-4e4b-11ea-8f7f-82fb702e6be8.png)

This PR would be just about perfect if we either update the validation method to allow a lack of those fields or save with `validate=False` https://github.com/ChildMindInstitute/mindlogger-app-backend/blob/ac4530037ec61f3dc08c626f6bff29213fce7d6e/girderformindlogger/models/model_base.py#L652-L661

(and of course [update the CHANGELOG](https://github.com/ChildMindInstitute/mindlogger-app-backend/blob/master/CONTRIBUTING.rst#submitting-a-pr))

Reply:
This actually works perfectly rebased onto a more recent commit: https://github.com/ChildMindInstitute/mindlogger-app-backend/commits/bcbf41e8615663a00ffac9cd663a658d59b6e499

![works perfectly](https://user-images.githubusercontent.com/5974438/74452087-94dabc00-4e4e-11ea-9c0a-025814d01e8c.png)